### PR TITLE
8303910: jdk/classfile/CorpusTest.java failed 1 of 6754 tests

### PR DESCRIPTION
**Pull request by asotona**
jdk/classfile/CorpusTest.java rarely fails in 1 of 6754 parametrised parallel junit tests

The root cause seems to be thread-unsafe lazy initialisation of `TypeKind.newarraycodeToTypeTag` in `TypeKind::fromNewArrayCode`. Provided patch replaces that lazy-initialized helper array with switch expression.

Please review.

Thanks,
Adam

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8303910](https://bugs.openjdk.org/browse/JDK-8303910): jdk/classfile/CorpusTest.java failed 1 of 6754 tests

### Reviewers
 * @liach (no known openjdk.org user name / role)
 * [Jaikiran Pai](https://openjdk.org/census#jpai) (@jaikiran - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/13004/head:pull/13004` \
`$ git checkout pull/13004`

Update a local copy of the PR: \
`$ git checkout pull/13004` \
`$ git pull https://git.openjdk.org/jdk pull/13004/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 13004`

View PR using the GUI difftool: \
`$ git pr show -t 13004`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/13004.diff">https://git.openjdk.org/jdk/pull/13004.diff</a>

</details>

### CODE REVIEW

File: src/java.base/share/classes/jdk/internal/classfile/TypeKind.java [hunk 1]
```diff
@@ -102,13 +100,17 @@ public TypeKind asLoadable() {
...
+        return switch (newarraycode) {
+            case 4 -> TypeKind.BooleanType;
+            case 5 -> TypeKind.CharType;
+            case 6 -> TypeKind.FloatType;
+            case 7 -> TypeKind.DoubleType;
+            case 8 -> TypeKind.ByteType;
+            case 9 -> TypeKind.ShortType;
+            case 10 -> TypeKind.IntType;
+            case 11 -> TypeKind.LongType;
+            default -> throw new IllegalStateException("Bad new array code: " + newarraycode);
```
Comment (liach) [inline src/java.base/share/classes/jdk/internal/classfile/TypeKind.java:112]: Should throw `IllegalArgumentException` to be more informative

Comment (asotona) [reply src/java.base/share/classes/jdk/internal/classfile/TypeKind.java]: Right, I've changed it on both places in `TypeKind`, thanks.

**Review by liach — APPROVED** [review]

**Review by jaikiran — APPROVED** [review]
Looks OK to me.

**Pull request closed without merging**